### PR TITLE
Preview: Simplify docsMode

### PR DESCRIPTION
**Pull request by tmeasday**
<strike>Telescoping on https://github.com/storybookjs/storybook/pull/18728</strike>

## What I did

- When in docs mode, only add entries of type `docs` from the index
- This massively simplifies all other logic so we can delete other code (also we don't need to mess with layout as docs entries will never show the toolbar etc)
- Also try and clean up references to the `DOCS_MODE` global and instead use props from the provider

## How to test

Run react-ts in docs mode and without.
 
We don't have any e2e tests for docs mode at this stage 😞

### CODE REVIEW

File: lib/api/src/modules/layout.ts [hunk 1]
```diff
@@ -73,11 +72,10 @@ export interface UIOptions {
 const defaultState: SubState = {
   ui: {
     enableShortcuts: true,
-    docsMode: false,
   },
   layout: {
     initialActive: ActiveTabs.CANVAS,
-    showToolbar: !DOCS_MODE,
+    showToolbar: true,
```
Comment (tmeasday) [inline lib/api/src/modules/layout.ts:78]: The toolbar will hide automatically as all entries are docs

File: lib/api/src/modules/stories.ts [hunk 2]
```diff
@@ -194,11 +195,6 @@ export const init: ModuleFn<SubAPI, SubState, true> = ({
       const { storiesHash, storyId, refs, refId } = store.getState();
       const story = api.getData(storyId, refId);
 
-      if (DOCS_MODE) {
-        api.jumpToComponent(direction);
-        return;
-      }
```
Comment (tmeasday) [inline lib/api/src/modules/stories.ts:200]: This happens naturally as all entries are docs, we don't have to skip over any.

File: lib/api/src/tests/layout.test.js [hunk 3]
```diff
@@ -10,7 +10,6 @@ beforeEach(() => {
   currentState = {
     ui: {
       enableShortcuts: true,
-      docsMode: false,
```
Comment (tmeasday) [inline lib/api/src/tests/layout.test.js:13]: We weren't using this

**Review by shilman — APPROVED** [review]
Great change! 🎉

**Pull request merged**